### PR TITLE
Simplify `JNLPLauncherRealTest` using the new methods in `InboundAgentRule`

### PR DESCRIPTION
**Pull request by Vlatombe**
This is a test update, downstream of https://github.com/jenkinsci/jenkins-test-harness/pull/517

### Proposed changelog entries

- N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7381"></a>

### CODE REVIEW

File: test/pom.xml
Comment (jglick) [inline test/pom.xml:96]: (in draft pending release)

File: test/src/test/java/hudson/slaves/JNLPLauncherRealTest.java
Comment (jglick) [inline test/src/test/java/hudson/slaves/JNLPLauncherRealTest.java:82]: Is this actually necessary, or can the `try` block be deleted and let the `@Rule` do its work?

**Review by jglick — APPROVED** [review]

File: test/src/test/java/hudson/slaves/JNLPLauncherRealTest.java
Comment (basil) [reply test/src/test/java/hudson/slaves/JNLPLauncherRealTest.java:82]: https://github.com/jenkinsci/jenkins-test-harness/pull/501#issuecomment-1284448230

Comment (jglick) [reply test/src/test/java/hudson/slaves/JNLPLauncherRealTest.java:82]: Thanks, I knew there was a discussion somewhere. Anyway, seems like once the apparent bug in production code is fixed, we _could_ delete these and let the rule cleanup run. So would be good to leave in a comment to that effect.

Comment (basil) [reply test/src/test/java/hudson/slaves/JNLPLauncherRealTest.java:82]: https://github.com/jenkinsci/jenkins-test-harness/pull/501#issuecomment-1284480345

**Review by jglick — APPROVED** [review]

**Review by basil — APPROVED** [review]
Thanks for the PR!

---

This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
